src/CarterExtensions.cs

```diff
@@ -13,6 +14,66 @@ namespace Carter
 
     public static class CarterExtensions
     {
+        /// <summary>
+        /// Adds Carter to the specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add Carter to.</param>
+        /// <param name="configure">Allows <see cref="CarterBootstrapper"/> to be configured.</param>
+        public static void AddCarter(this IServiceCollection services, Action<CarterBootstrapper> configure)
+        {
+            // hack to test without changing all tests
+            // AddCarterScoped(services);
+            // return;
+            
+            var bootstrapper = new CarterBootstrapper();
+            configure(bootstrapper);
+            if (bootstrapper.ResponseNegotiators.All(x => x.GetType() != typeof(DefaultJsonResponseNegotiator)))
+            {
+                bootstrapper.RegisterResponseNegotiators(new DefaultJsonResponseNegotiator());    
+            }            
+            services.AddSingleton<IValidatorLocator>(new CarterValidatorLocator(bootstrapper.Validators));
+            services.AddSingleton<IEnumerable<IResponseNegotiator>>(bootstrapper.ResponseNegotiators);
+            services.AddSingleton<ICarterBootstrapper>(bootstrapper);
+            services.AddRouting();
+        }
+
+        /// <summary>
+        /// Adds Carter to the specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add Carter to.</param>
+        /// <param name="configure">Allows <see cref="CarterBootstrapper"/> to be configured.</param>
+        public static void AddCarterScoped(this IServiceCollection services)
```

Comment:
Why `AddCarterScoped` and not just `AddCarter`?  Not really a fan of this method name.

Reply:
Yup should be the same